autopush_rs/src/server/mod.rs

```diff
@@ -270,17 +273,20 @@ impl Server {
 
             // Internal HTTP server setup
             {
-                use hyper::server::Http;
-
                 let handle = core.handle();
                 let router_ip = resolve(&srv.opts.router_ip);
                 let addr = format!("{}:{}", router_ip, srv.opts.router_port)
                     .parse()
                     .unwrap();
                 let push_listener = TcpListener::bind(&addr, &handle).unwrap();
-                let proto = Http::new();
-                let push_srv = push_listener.incoming().for_each(move |(socket, addr)| {
-                    proto.bind_connection(&handle, socket, addr, ::http::Push(srv.clone()));
+                let http = Http::<hyper::Chunk>::new();
+                let push_srv = push_listener.incoming().for_each(move |(socket, _)| {
```

Comment:
I take it this defaults to addr '0.0.0.0'?

Reply:
Above it binds to the addr derived from router_ip/port (AutopushConfig's router_hostname). I'm not sure that's quite right actually

Reply:
https://github.com/mozilla-services/autopush/issues/1113